src/Language/Haskell/LSP/Test/Decoding.hs

```diff
@@ -120,6 +120,7 @@ matchResponseMsgType req = case req of
   where decoded x = fromMaybe (error $ "Couldn't decode response for the request type: "
                                         ++ show req ++ "\n" ++ show x)
                               (decode x)
+        removeNullResult x = maybe x (<> "}") (B.stripSuffix ",\"result\":null}" x)
```

Comment:
An ugly hack to sanitize the response from the manual javascript session
Maybe we should have a generic way to remove fields with null values or change the json encoding to make `field:null` = `_field=Nothing`

Reply:
I wouldn't worry about the javascript test too much. This looks like its caused by how haskell-lsp [doesn't parse responses with null results](https://github.com/alanz/haskell-lsp/issues/162) (which should probably be fixed at some point). You can just revert this and let the javascript test fail for now